### ISSUE TITLE
Figure out what's up with Mojave

### ISSUE THREAD

**Issue by adminspotter** [adminspotter/cuddly-gl#73]
The most recent MacOS Mojave upgrade changes some things about its GL stack, and our test program doesn't work quite the same any longer.

First, we're getting warnings with a couple of the functions in our `uitest` double-buffered drawing loop.

```plain
g++ -DHAVE_CONFIG_H -I. -I..    -I/Users/tquirk/include -DGLM_FORCE_RADIANS -I/opt/X11/include/freetype2  -std=gnu++11 -I/opt/X11/include -I/Users/tquirk/include -I/Users/tquirk/include -DGLM_FORCE_RADIANS -I/opt/X11/include/freetype2  -std=gnu++11 -I/opt/X11/include -MT uitest-uitest.o -MD -MP -MF .deps/uitest-uitest.Tpo -c -o uitest-uitest.o `test -f 'uitest.cc' || echo './'`uitest.cc
uitest.cc:245:9: warning: 'glClearColor' is deprecated: first deprecated in
      macOS 10.14 - OpenGL API deprecated. (Define GL_SILENCE_DEPRECATION to
      silence these warnings) [-Wdeprecated-declarations]
        glClearColor(0.0f, 0.0f, 0.0f, 1.0f);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Headers/gl.h:2396:13: note: 
      'glClearColor' has been explicitly marked deprecated here
extern void glClearColor (GLclampf red, GLclampf green, GLclampf blue, G...
            ^
uitest.cc:246:9: warning: 'glClear' is deprecated: first deprecated in macOS
      10.14 - OpenGL API deprecated. (Define GL_SILENCE_DEPRECATION to silence
      these warnings) [-Wdeprecated-declarations]
        glClear(GL_COLOR_BUFFER_BIT);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Headers/gl.h:2394:13: note: 
      'glClear' has been explicitly marked deprecated here
extern void glClear (GLbitfield mask) OPENGL_DEPRECATED(10.0, 10.14);
            ^
2 warnings generated.
```

`glClearColor` and `glClear` are deprecated.  We could set the `GL_SILENCE_DEPRECATION` flag, but that might just generate a pretty big surprise for us later on.  It appears that the functions were deprecated in OpenGL 3.0 and removed in OpenGL 3.1, but are retained with a compatibility profile.<sup>[1]</sup>  So it's very likely that Mojave isn't the only platform that will have to deal with this.  Searching suggests that `glClearBufferfv` would be the right replacement, but all examples I can find use the deprecated functions, and the manual pages for the replacement function are... less than completely clear.

Secondly, and this may be related to the `glClear` business above, when first running `uitest`, the window is black.  Simply moving the window seems to kick-start all the drawing to work properly, and all the widgets function as expected.

[1] https://www.opengl.org/discussion_boards/archive/index.php/t-182617.html

**Comment by adminspotter** [comment]
We found an article that said we should be including `<OpenGL/gl3.h>` on MacOS, instead of `<GL/gl.h>`.  We tried adding some conditional `#ifdef __APPLE__` around all inclusions of the relevant headers, and now a whole ton more functions are marked deprecated, and we have also picked up some errors.  So that's no solution.

Also checked some documentation for `glClearColor` and `glClear`, and they are not deprecated in any GL versions up through 4.5.  So this appears to be a case of MacOS taking gross liberties with what's actually deprecated and what's not.  Thanks a ton, Apple. :(

If we weren't seeing the weird black screen to start (which started at the same time, so I'm assuming that they are at least somewhat related), I would close this as won't-fix and move on.
